### PR TITLE
Add support for SQLAlchemy 2.0, continued

### PR DESCRIPTION
**Pull request by mhauru**
Continued from #249, see discussion there.

### CODE REVIEW

**Comment by agronholm** [comment]
Thanks, I'll check this out later today.

**Review by agronholm — CHANGES_REQUESTED** [review]
I think the code is good enough for merging. I'll do some tinkering on it later.
Aside from the suggestions I've added, I will still need an added changelog entry (remember to give credit to yourself and the author of the original PR).

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -20,8 +21,12 @@ jobs:
         python-version: ${{ matrix.python-version }}
```
[the remark this answers is not in the document]
Comment (mhauru) [reply .github/workflows/test.yml:21]: Done

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -10,7 +10,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.7", "3.9", "3.11"]
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
```
[the remark this answers is not in the document]
Comment (mhauru) [reply .github/workflows/test.yml]: Done

File: pyproject.toml [hunk 3]
```diff
@@ -88,7 +91,7 @@ show_missing = true
 [tool.tox]
 legacy_tox_ini = """
 [tox]
-envlist = py37, py38, py39, py310
+envlist = py37, py38, py39, py310, py311
```
[the remark this answers is not in the document]
Comment (mhauru) [reply pyproject.toml]: Done

**Comment by mhauru** [comment]
Tests fail on 3.12, but I made the other changes. Please edit the changelog entry to be to your liking.

File: .github/workflows/test.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (agronholm) [reply .github/workflows/test.yml:13]: Greenlet 3.0.0a1 works with Python 3.12. I've worked around this in another project like this: https://github.com/agronholm/apscheduler/blob/master/pyproject.toml#L46

**Review by agronholm — APPROVED** [review]

**Comment by agronholm** [comment]
Thanks a lot!

**Comment by mhauru** [comment]
No problem, @rbuffat did all the actual work here. Do you plan to release this to pypi or do you want further changes before that?

**Comment by agronholm** [comment]
I think I will first tinker with the test suite and then release to PyPI later this week.

**Comment by amacfie** [comment]
I'm not getting the `nullable` parameter in the `mapped_column` calls when the column is `NOT NULL`

**Comment by agronholm** [comment]
What are you getting exactly?

**Comment by amacfie** [comment]
Nevermind I guess it's not necessary <https://docs.sqlalchemy.org/en/20/changelog/whatsnew_20.html#:~:text=All%20nullable%20parameters%20can%20be%20implied%20using%20Optional%5B%5D>

**Comment by mhauru** [comment]
Hi @agronholm, sorry to keep pestering you, I know you have competing priorities. Is there anything I can do to help in getting the SQLAlchemy 2 support into a PyPi release? It would unblock a series of dependency upgrades for our project.

**Comment by agronholm** [comment]
I wanted to fix a couple more bugs first, but looks like I didn't have the time. I've released v3.0.0rc3 now to unblock your upgrades.

**Comment by pang-thebasement** [comment]
Hello:
Thanks a lot to all contributors. I wanted to test sqlacodegen on our database, but when I try to pip install tag 3.0.0rc3

`git checkout 3.0.0rc3`
`python3 -m pip install -U .`

I get an error like this:

        Defaulting to user installation because normal site-packages is not writeable
        Processing /home/myuser/extra/software/sqlacodegen
          Installing build dependencies ... done
          Getting requirements to build wheel ... done
          Installing backend dependencies ... done
          Preparing metadata (pyproject.toml) ... done
        Building wheels for collected packages: UNKNOWN
          Building wheel for UNKNOWN (pyproject.toml) ... done
          Created wheel for UNKNOWN: filename=UNKNOWN-3.0.0rc3-py3-none-any.whl size=1867 sha256=d4d18a167a8f7c8f5c6a2bb9d8703aee259487e55336fc1e475998db872a5a99
          Stored in directory: /tmp/pip-ephem-wheel-cache-pj6osbkb/wheels/c5/88/d7/6594ab3afff35313f447a11f0f0dfe560bffee6b30ace33111
        Successfully built UNKNOWN
        Installing collected packages: UNKNOWN
        Successfully installed UNKNOWN-3.0.0rc3

After which there is a python package actually called UNKNOWN, which I can uninstall with `python3 -m pip uninstall UNKNOWN`
Sorry I can't help. The `pyproject.toml` looks fine to me. I haven't really used that format :-/

**Comment by pang-thebasement** [comment]
Thanks, that works! And it gave in a blink some nice looking python code! Genius!

**Pull request merged**